test/Libraries/Microsoft.Extensions.DataIngestion.Tests/Chunkers/OverlapDocumentTokenChunkerTests.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.ML.Tokenizers;
+using Xunit;
+
+namespace Microsoft.Extensions.DataIngestion.Chunkers.Tests
+{
+    public class OverlapDocumentTokenChunkerTests : DocumentTokenChunkerTests
+    {
+        protected override IngestionChunker<string> CreateDocumentChunker(int maxTokensPerChunk = 2_000, int overlapTokens = 500)
+        {
+            var tokenizer = TiktokenTokenizer.CreateForModel("gpt-4o");
+            return new DocumentTokenChunker(new(tokenizer) { MaxTokensPerChunk = maxTokensPerChunk, OverlapTokens = overlapTokens });
+        }
+
+        [Fact]
+        public async Task TokenChunking_WithOverlap()
+        {
+            string text = "The quick brown fox jumps over the lazy dog";
+            var tokenizer = TiktokenTokenizer.CreateForModel("gpt-4o");
+            int chunkSize = 4;  // Small chunk size to demonstrate overlap
+            int chunkOverlap = 1;
+
+            var chunker = new DocumentTokenChunker(new(tokenizer) { MaxTokensPerChunk = chunkSize, OverlapTokens = chunkOverlap });
```

Comment:
nit: should we just use `CreateDocumentChunker` here?
```suggestion
            string text = "The quick brown fox jumps over the lazy dog";
            var chunker = CreateDocumentChunker(maxTokensPerChunk: 4, chunkOverlap: 1);
```

Reply:
No, because I wanted to be explicit here. I've removed only the `chunkOverlap` variable. We need to use the tokenizer within the scope of method and I don't want to rely on the same one being used in `CreateDocumentChunker` even if that's the case now. Similarly the `chunkSize` is used in some assertions.